### PR TITLE
daml ledger: add a max-inbound-message-size flag for grpc

### PR DESCRIPTION
**Pull request by robin-da**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-assistant/daml-helper/src/DA/Daml/Helper/Main.hs [hunk 1]
```diff
@@ -367,6 +369,12 @@ commandParser = subparser $ fold
         <> metavar "TOKEN_PATH"
         <> help "Path to the token-file for ledger authorization"
 
+    maxInboundSizeFlag :: Parser (Maybe Natural)
+    maxInboundSizeFlag = optional . option auto $
+        long "max-inbound-message-size"
+        <> metavar "INT"
+        <> help "Maximal size of inbound messages via gRPC in bytes"
```
Comment (cocreature) [inline daml-assistant/daml-helper/src/DA/Daml/Helper/Main.hs:376]:
```suggestion
        <> help "Optional max inbound message size in bytes. This only affect connections via gRPC."
```
A bit closer to the description we use everywhere else.

**Review by cocreature — APPROVED** [review]
Nice! A test might be good.

**Comment by robin-da** [comment]
@cocreature do we already have test infrastructure to test with big dar files somewhere?

**Comment by cocreature** [comment]
Not sure we do but I’d go the other way around. Lower the limit and assert on the failure.

File: daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs [hunk 2]
```diff
@@ -119,6 +125,7 @@ getDefaultArgs LedgerFlags { fApi
       , tokM = tokM
       , timeout = fTimeout
       , sslConfigM = fSslConfigM
+      , grpcArgs = [MaxReceiveMessageLength n | Just n <- [fMaxReceiveLengthM]]
```
Comment (hurryabit) [inline daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs:128]: Maybe that's easier to understand, but I'm not sure:
```suggestion
      , grpcArgs = MaxReceiveMessageLength <$> toList fMaxReceiveLengthM
```

**Review by hurryabit — COMMENTED** [review]
Nice. Thank you.

**Pull request merged**